libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/internal/MapboxTripNotification.kt

```diff
@@ -212,12 +211,11 @@ class MapboxTripNotification constructor(
      * @param applicationContext the application's [Context]
      * @return [PendingIntent] to opening application
      */
-    @SuppressLint("UnspecifiedImmutableFlag")
     private fun createPendingOpenIntent(applicationContext: Context): PendingIntent? {
         val pm = applicationContext.packageManager
         val intent = pm.getLaunchIntentForPackage(applicationContext.packageName) ?: return null
         intent.setPackage(null)
-        return PendingIntent.getActivity(applicationContext, 0, intent, 0)
+        return PendingIntent.getActivity(applicationContext, 0, intent, PendingIntent.FLAG_IMMUTABLE)
```

Comment:
[PendingIntent.FLAG_IMMUTABLE](https://developer.android.com/reference/android/app/PendingIntent#FLAG_IMMUTABLE) was added in API version 23. Our min version is 21.

Reply:
I tested it in an Android 5.0 (API level 21) and didn't run into any problems. Also checked the documentation and didn't see that it's marked as not available in previous versions. At the end of the day it's an `int`, I think we should be fine.

Reply:
Android Studio gives a warning here:
<img width="1496" alt="Screenshot 2021-11-12 at 15 41 33" src="https://user-images.githubusercontent.com/2395284/141468919-45f5bbe4-bd0f-4857-ae72-6494004b1bb4.png">
I know it is a compile time constant, so this won't lead to `NoClassDefFoundError` exceptions or anything like that, but in theory there can be a problem of using unknown flags. You can't just pass any int to the method, because it is a bit mask. And simple version check wouldn't hurt.

Reply:
Yeah correct. I also saw it 😅

Reply:
Fixed it in https://github.com/mapbox/mapbox-navigation-android/pull/5102/commits/bf818e0f712dc1e26c70cb4e7efcb8eee9a946ae